crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -220,8 +236,16 @@ pub struct FeeFactor(f64);
 
 impl FeeFactor {
     /// Convert a fee into a `FeeFactor` capping its value
-    pub fn try_from_capped(value: f64, cap: f64) -> anyhow::Result<Self> {
-        value.max(0.0).min(cap).try_into()
+    pub fn try_from_capped(
+        value: Decimal,
+        cap: Decimal,
+        accumulated: Decimal,
+    ) -> anyhow::Result<Self> {
+        // Calculate how much more we can add before hitting the cap
+        let remaining_factor = (Decimal::ONE + cap) / (Decimal::ONE + accumulated) - Decimal::ONE;
```

Comment:
Took me a while to figure out why this is correct and why `Decimal::ONE` is needed. A comment explaining that `cap` starts at 0 (i.e. 30% cap == 0.3) whereas the math only works if it starts at 1 (i.e. 30% cap == 1.3) would be helpful.